### PR TITLE
BF: Achieve Numpy 1.10 compatibility (fixes gh-381)

### PR DESCRIPTION
**Pull request by hanke**
Casting rules changed.

### CODE REVIEW

**Comment by hanke** [comment]
Tests are happy. Please merge @nno @yarikoptic

File: mvpa2/measures/anova.py [hunk 1]
```diff
@@ -79,12 +79,12 @@ def _call(self, dataset):
         ul = targets_sa.unique
 
         na = len(ul)
-        bign = float(dataset.nsamples)
         alldata = dataset.samples
 
         # total squares of sums
         sostot = np.sum(alldata, axis=0)
         sostot *= sostot
+        bign = sostot.dtype.type(dataset.nsamples)
```
Comment (yarikoptic) [inline mvpa2/measures/anova.py]: this doesn't sound right -- if alldata.dtype is int, you will end up with int here, which would also point to the bug below that inplace division

```
$> python -c 'import numpy as np; a =np.arange(8).reshape((2, -1)); a = np.sum(a, axis=0); a*= a; print a.dtype.type; b=a.copy();  a /= 3; print a; print b/float(3)' 
<type 'numpy.int64'>
[ 5 12 21 33]
[  5.33333333  12.          21.33333333  33.33333333]
```
and I guess test isn't precise to care about those rounding effects

Comment (hanke) [inline mvpa2/measures/anova.py]: Ah, right. Didn't think about that. So we are left with a full copy on div...

Comment (yarikoptic) [inline mvpa2/measures/anova.py]: On Mon, 11 Jan 2016, Michael Hanke wrote:
>  >          alldata = dataset.samples

>  >          # total squares of sums
>  >          sostot = np.sum(alldata, axis=0)
>  >          sostot *= sostot
>  > +        bign = sostot.dtype.type(dataset.nsamples)

>    Ah, right. Didn't think about that. So we are left with a full copy on
>    div...

only if type is not float ;)

**Comment by hanke** [comment]
Will merge this in -- have a bunch of fixes lined up and this is holding up the progress. Will deal with the fallout, if any.

**Comment by yarikoptic** [comment]
Was it fixed for my concern of incorrect results if type was interested?

**Comment by yarikoptic** [comment]
Bloody phone. int not interested

**Pull request merged**